src/pages/PublicAppointments/PatientRegistration.tsx

```diff
@@ -208,13 +211,14 @@ export function PatientRegistration(props: PatientRegistrationProps) {
       geo_organization: data.geo_organization,
       is_active: true,
     };
+    setIsSubmitting(true);
     createPatient(formattedData);
   });
 
   // TODO: Use useBlocker hook after switching to tanstack router
   // https://tanstack.com/router/latest/docs/framework/react/guide/navigation-blocking#how-do-i-use-navigation-blocking
   useNavigationPrompt(
-    form.formState.isDirty && !isCreatingPatient,
+    form.formState.isDirty && !isCreatingPatient && !isSubmitting,
```

Comment:
Swap out isCreatingPatient with isCreatingAppointment; removes the need for an extra state 👍